modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java

```diff
@@ -0,0 +1,186 @@
+                    Assert.assertEquals((String) object.get("name"), "Marketing");
+
+                    //store the create category's UUID to later use in the category update test
+                    categoryId = (String) object.get("id");
+                } finally {
+                    response.close();
+                }
+            }
+        }
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test add API category with duplicate name", dependsOnMethods = {
+            "testAddAPICategory" })
+    public void addAPICategoryWithDuplicateName() throws Exception {
+        try (CloseableHttpClient client = HTTPSClientUtils.getHttpsClient();) {
+            HttpPost post = new HttpPost(categoriesAdminAPIURL);
+            post.addHeader(APIMIntegrationConstants.AUTHORIZATION_HEADER,
+                    "Basic " + encodeCredentials(user.getUserName(), user.getPassword().toCharArray()));
+            post.addHeader("Content-Type", "application/json");
+            StringEntity payload = new StringEntity(
+                    "{\"name\": \"Marketing\", \"description\": \"Marketing category\"}", "UTF-8");
+            payload.setContentType("application/json");
+            post.setEntity(payload);
+            CloseableHttpResponse response = client.execute(post);
+            try {
+                HttpEntity entity = response.getEntity();
+                String responsePayload = EntityUtils.toString(entity);
+                Assert.assertTrue(responsePayload.contains("Category with name 'Marketing' already exists"));
+                Assert.assertEquals(response.getStatusLine().getStatusCode(), 500);
+            } finally {
+                response.close();
+            }
+        }
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test update API category", dependsOnMethods = {
+            "addAPICategoryWithDuplicateName" })
+    public void testUpdateAPICategory() throws Exception {
+        try (CloseableHttpClient client = HTTPSClientUtils.getHttpsClient();) {
+            HttpPut put = new HttpPut(categoriesAdminAPIURL + "/" + categoryId);
+            put.addHeader(APIMIntegrationConstants.AUTHORIZATION_HEADER,
+                    "Basic " + encodeCredentials(user.getUserName(), user.getPassword().toCharArray()));
+            put.addHeader("Content-Type", "application/json");
+            StringEntity payload = new StringEntity("{\"name\": \"Sales\", \"description\": \"Sales category\"}",
+                    "UTF-8");
+            payload.setContentType("application/json");
+            put.setEntity(payload);
+            try (CloseableHttpResponse response = client.execute(put);) {
+                HttpEntity entity = response.getEntity();
+                Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+            }
+        }
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test delete API category",
+            dependsOnMethods = { "testAddAPICategory" })
```

Comment:
Let's change this depends on to "testUpdateAPICategory"

Reply:
fuxed in https://github.com/wso2/product-apim/pull/7245/commits/01de486f9a550d1f94acf9bb03fd024128f46e3f